code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -6302,16 +6302,22 @@
 	dupeable = FALSE
 
 /datum/reagent/ethanol/scientists_serendipity/handle_special_behavior(var/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/D)
-	if(volume<10)
-		glass_icon_state = "scientists_surprise"
-		glass_name = "\improper Scientist's Surprise"
-		glass_desc = "There is as yet insufficient data for a meaningful answer."
-	else
-		glass_icon_state = "scientists_serendipity"
-		glass_name = "\improper Scientist's Serendipity"
-		glass_desc = "Knock back a cold glass of R&D."
-		D.origin_tech = "materials=7;engineering=3;plasmatech=2;powerstorage=4;bluespace=6;combat=3;magnets=6;programming=3"
-
+		switch(volume)
+			if(1 to 9)
```

Comment:
`volume` isn't an integer. This should be `0 to 10` and the other one should also go all the way up to the bottom of the next. Otherwise nothing would happen if it were between 9 and 10 units or between 49 and 50 units.

Reply:
>This should be 0 to 10 and the other one should also go all the way up to the bottom of the next. Otherwise nothing would happen if it were between 9 and 10 units or between 49 and 50 units.

So it should be if(0 to 10), if(10 to 50) and if (50 to INFINITY)?

>Oh and the order should probably be reversed so you hit it if you get the minimum instead of just above the minimum

As in putting the if conditions in descending order or putting it like if(50 to 10)?

Reply:
Also, the original Serendipity code had "if(volume >10)" as it's volume check. I wanted to use it but didn't know how, so I went with "if(10 to 50)".

Reply:
For your first question, yes. For your second question, no, reversed as in put the `50 to INFINITY` one first and the `0 to 10` one last.
Yes, you can do `(volume > 10) && (volume < 50)`

Reply:
Is this better?